pkg/util/azureclient/applens/applens_client.go

```diff
@@ -0,0 +1,163 @@
+package applens
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+// AppLens Client created from CosmosDB Client
+// (https://github.com/Azure/azure-sdk-for-go/blob/3f7acd20691214ef2cb1f0132f82115f1df01a8c/sdk/data/azcosmos/cosmos_client.go)
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/gofrs/uuid"
+)
+
+// AppLens client is used to interact with the Azure AppLens service.
+type Client struct {
+	endpoint string
+	pipeline runtime.Pipeline
+}
+
+// Endpoint used to create the client.
+func (c *Client) Endpoint() string {
+	return c.endpoint
+}
+
+// NewClient creates a new instance of AppLens client with Azure AD access token authentication. It uses the default pipeline configuration.
+// endpoint - The applens service endpoint to use.
+// cred - The credential used to authenticate with the applens service.
+// options - Optional AppLens client options.  Pass nil to accept default values.
+func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (*Client, error) {
+	scope, err := createScopeFromEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{endpoint: endpoint, pipeline: newPipeline([]policy.Policy{runtime.NewBearerTokenPolicy(cred, scope, nil), &appLensBearerTokenPolicy{}}, o)}, nil
+}
+
+func newPipeline(authPolicy []policy.Policy, options *ClientOptions) runtime.Pipeline {
+	if options == nil {
+		options = &ClientOptions{}
+	}
+
+	return runtime.NewPipeline("applens", serviceLibVersion,
+		runtime.PipelineOptions{
+			PerCall:  []policy.Policy{},
+			PerRetry: authPolicy,
+		},
+		&options.ClientOptions)
+}
+
+func createScopeFromEndpoint(endpoint string) ([]string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return []string{fmt.Sprintf("%s://%s/.default", u.Scheme, u.Hostname())}, nil
```

Comment:
i'm pretty sure the endpoint is not the scope.

Reply:
This is a carry over from the Cosmos client - I figured I would resolve any issues related to the scope during testing as I wasn't sure whether it would be needed or not. Do you think the scope should be something else, or removed altogether?

Reply:
you have to have scope to get AAD token. in ADAL, it's called ResourceID. In MSAL, it's called Scope.
https://msazure.visualstudio.com/CloudNativeCompute/_git/aks-rp?path=/resourceprovider/server/microsoft.com/containerservice/templates/implementation.go&version=GBmaster&line=676&lineEnd=677&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

Reply:
Understood - with that information I was able to use the appropriate scopes provided by the AppLens team. The code for creating the scope from the target endpoint has been removed.